src/p5.dance.interpreted.js

```diff
@@ -178,14 +178,31 @@ function whenPeak(range, func) {
 }
 
 /**
+ * Execute code at a specified time, either in measures or seconds.
+ * One idiosyncrasy: despite the function's name, if we call atTimestamp(4, "measures", foo),
+ * we actually fire on the 5th measure (i.e after 4 measures have completed).
  * @param {number} timestamp
  * @param {string} unit - Should be "measures" or "seconds"
  * @param {function} func - Code to run when event fires
  */
 function atTimestamp(timestamp, unit, func) {
-  // Despite the functions name, if we call atTimestamp(4, "measures", foo), we
-  // actually want to fire on the 5th measure (i.e after 4 measures have completed)
-  if (unit === 'measures') {
+  atTimestampEventGenerator(timestamp, unit, func, true);
+}
+
+/**
+ * Execute code at a specified time, either in measures or seconds.
+ * Has "NotAfter" suffix to distinguish from "atTimestamp",
+ * which executes code **AFTER** x measures (if time is specified in measures).
+ * @param {number} timestamp
+ * @param {string} unit - Should be "measures" or "seconds"
+ * @param {function} func - Code to run when event fires
+ */
+function atTimestampNotAfter(timestamp, unit, func) {
```

Comment:
Haha this is a tough one 🙂 I don't mind this name, maybe `atTimestampOrMeasure`/`onTimestampOrMeasure` as another idea?

Reply:
When I did this yesterday I was thinking that I couldn't change the existing function name for legacy reasons, but realizing today I think that's not true (ie, we only serialize the name of the block, so we can update the block definition to call a different function and things will still work fine)? @fisher-alice can you confirm my understanding?

To align with the text that actually appears in the blocks, I could change the existing function to `afterTimestamp` and make the new one `atTimestamp`?

Reply:
Yes I believe that's correct. You could change the `func` property in the 'after x measures' block config file after updating the function name. That would be so much less confusing!

Reply:
I just looked and the name of the `json` file is `Dancelab_atTimeStamp.json` - so we wouldn't change the name of this file - is that correct? So still kinda confusing to a reader.

Reply:
Oh right, the block name can't be changed. So maybe just leave as-is 🤦